tests/foreman/api/test_registration.py

```diff
@@ -262,3 +262,61 @@ def test_negative_capsule_without_registration_enabled(
         "Proxy lacks one of the following features: 'Registration', 'Templates'"
         in context.value.response.text
     )
+
+
+@pytest.mark.rhel_ver_match('[^6]')
+def test_positive_host_registration_with_non_admin_user(
+    module_org,
+    module_location,
+    module_activation_key,
+    module_target_sat,
+    rhel_contenthost,
+):
+    """Verify host registration with non admin user
+
+    :id: 02bdda6a-010d-4098-a7e0-e4b5e8416ce3
+
+    :steps:
+        1. Enable the required repositories on the satellite and get a host(RHEL7/RHEL8/RHEL9)
+        2. Create a non-admin user and assign "Register Hosts" role to it.
+        3. Create an activation key and assign CV and LCE to it.
+        4. Login Satellite with new user and generate curl command to register host with all setup options set to NO.
+        5. Run the curl command on the host.
+
+    :expectedresults: Host registered successfully with all setup options set to 'NO' for non admin user
+
+    :BZ: 2211484
+
+    :customerscenario: true
+    """
+    register_host_role = module_target_sat.api.Role().search(
+        query={'search': 'name="Register hosts"'}
+    )
+    module_target_sat.api.User(
+        role=register_host_role,
+        admin=False,
```

Comment:
@amolpati30 I just realized we're not using this created non-admin user while generating a register command, and the main intent of this test is to verify non-admin user as per linked BZ?

Reply:
@amolpati30 just for practice, resolve comments only when either you reply for it if you disagree or fix this in your commits

Reply:
Changes has been done. I have created non-admin user while generating a register command.